pkg/query-service/app/integrations/messagingQueues/kafka/sql.go

```diff
@@ -318,6 +319,141 @@ GROUP BY
 	return query
 }
 
+// generateOverviewSQL builds the ClickHouse SQL query with optional filters.
+// If a filter slice is empty, the query does not constrain on that field.
+func generateOverviewSQL(start, end int64, filters *QueueFilters) string {
+	// Convert from nanoseconds to float seconds in Go to avoid decimal overflow in ClickHouse
+	startSeconds := float64(start) / 1e9
+	endSeconds := float64(end) / 1e9
+
+	// Compute time range difference in Go
+	timeRangeSecs := endSeconds - startSeconds
+
+	// Example ts_bucket boundaries (could be your own logic)
+	tsBucketStart := startSeconds - 1800
+	tsBucketEnd := endSeconds
+
+	// Build WHERE clauses for optional filters
+	// We always require messaging_system IN ('kafka', 'celery'), but
+	// we add additional AND conditions only if the slices are non-empty.
+	var whereClauses []string
+
+	// Mandatory base filter: show only kafka/celery
+	whereClauses = append(whereClauses, "aggregated_metrics.messaging_system IN ('kafka', 'celery')")
+
+	if len(filters.ServiceName) > 0 {
+		whereClauses = append(whereClauses, inClause("aggregated_metrics.service_name", filters.ServiceName))
+	}
+	if len(filters.SpanName) > 0 {
+		whereClauses = append(whereClauses, inClause("aggregated_metrics.span_name", filters.SpanName))
+	}
+	if len(filters.Queue) > 0 {
+		// "queue" in the struct refers to the messaging_system in the DB
+		whereClauses = append(whereClauses, inClause("aggregated_metrics.messaging_system", filters.Queue))
+	}
+	if len(filters.Destination) > 0 {
+		whereClauses = append(whereClauses, inClause("aggregated_metrics.destination", filters.Destination))
+	}
+	if len(filters.Kind) > 0 {
+		whereClauses = append(whereClauses, inClause("aggregated_metrics.kind_string", filters.Kind))
+	}
+
+	// Combine all WHERE clauses with AND
+	whereSQL := strings.Join(whereClauses, "\n    AND ")
+
+	// Final query string
+	// Note the use of %f for float64 values in fmt.Sprintf
+	query := fmt.Sprintf(`
+WITH
+    timeRange AS (
+        SELECT %f AS seconds
+    ),
+    processed_traces AS (
+        SELECT
+            resource_string_service$$name AS service_name,
+            name AS span_name,
+            CASE
```

Comment:
Why is there a need for this case? The feature should work for any `messaging.system` and not just kafka and celery

Reply:
This would work for other `messaging.system`, yes if the semantic conventions are followed.

But, I'm not sure what all instrumentations are supported for queues, this check here ensures that it works for celery and kafka without exception.

We can extend this by having a default case for any messaging queue, but probably not now, unless we have checked upstream instrumentations for different queues.

Reply:
The PRD was designed considering all messaging queues. Please remove this restriction. There are many other messaging systems which will start showing up automatically in the UI and will help users

Reply:
the table only needs `messaging.system` and `messaging.destination` which are there for many other systems

Reply:
makes sense, updated it with
`WHEN attribute_string_messaging$$system != '' THEN attribute_string_messaging$$system` for celery let's keep it same.